examples/sglang/README.md

```diff
@@ -242,3 +242,42 @@ DeepGEMM kernels can sometimes take a while to warm up. Here we provide a small
 ```bash
 ./warmup.sh HEAD_PREFILL_NODE_IP
 ```
+
+###### Benchmarking Disaggregated with WideEP
+
+In the official [blog post repro instructions](https://github.com/sgl-project/sglang/issues/6017), SGL uses batch inference to benchmark their prefill and decode workers. They do this by pretokenizing the ShareGPT dataset and then creating a batch of 8192 requests with ISL 4096 and OSL 5 (for prefill stress test) and a batch of 40000 with ISL 2000 and OSL 100 (for decode stress test). If you want to repro these benchmarks, you will need to uncomment the labeled flags in the `configs/dsr1.yaml` file inside of the container.
+
+We currently provide 2 different ways to perform an end to end benchmark which includes using our OpenAI frontend and tokenization. We will continue to add better support for these sorts of large single batch workloads in the future.
+
+> [!NOTE]
+> In order to use GenAI-Perf you will need to install `genai-perf`. You can find the instructions [here](https://github.com/triton-inference-server/perf_analyzer/tree/main/genai-perf). Once installed, you can run the `bench.sh` script below
+
+1. **GenAI Perf to benchmark end to end performance with 8k ISL 256 OSL**
+We've found that 8k ISL 256 OSL provides a good baseline for measuring end to end disaggregated serving performance for DSR1. As WideEP allows for a higher throughput, we provide a script that runs this workload at high concurrencies. DeepGEMM kernels can sometimes take a while to warm up. We provide a short ramping warmup script that can be used.
+
+Example usage:
+```bash
+# warmup
+./utils/bench.sh HEAD_PREFILL_NODE_IP --type warmup
+# run benchmark
+./utils/bench.sh HEAD_PREFILL_NODE_IP --type e2e
+```
+
+1. **GenAI Perf to benchmark completions with custom dataset**
+We provide a script that generates a JSONL file of the ShareGPT dataset and then use GenAI Perf to benchmark the prefill and decode workers. We use ShareGPT in order to leverage the pre-existing EPLB distributions provided by the SGLang team. If you don't want to use ShareGPT - you can also use GenAIPerf's synthetic dataset setup But note you will have to use dynamic EPLB configurations or record your own as the `init-expert-location` provided by SGLang is tuned specifically for the ShareGPT dataset at a 4096 ISL and 5 OSL.
+
+Example usage:
+```bash
+# generate data
+python3 utils/generate_bench_data.py --output data.jsonl --num-prompts 100 --input-len 40 --output-len 5 --model deepseek-ai/DeepSeek-R1
```

Comment:
put real ISL/OSL values for "reproducing" blog post results in the example commands